File: backend/src/main/java/com/rstreamdev/iamhungry/domain/RecipeBasic.java
package com.rstreamdev.iamhungry.domain;

import java.io.Serializable;

public class RecipeBasic implements Serializable {

    private int id;
    private String title;
    private String image;
    private int usedIngredientCount;
    private int missedIngredientCount;
    private int likes;

    public RecipeBasic() { }

    public RecipeBasic(int id, String title, String image, int usedIngredientCount, int missedIngredientCount, int likes) {
        this.id = id;
        this.title = title;
        this.image = image;
        this.usedIngredientCount = usedIngredientCount;
        this.missedIngredientCount = missedIngredientCount;
        this.likes = likes;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public int getUsedIngredientCount() {
        return usedIngredientCount;
    }

    public void setUsedIngredientCount(int usedIngredientCount) {
        this.usedIngredientCount = usedIngredientCount;
    }

    public int getMissedIngredientCount() {
        return missedIngredientCount;
    }

    public void setMissedIngredientCount(int missedIngredientCount) {
        this.missedIngredientCount = missedIngredientCount;
    }

    public int getLikes() {
        return likes;
    }

    public void setLikes(int likes) {
        this.likes = likes;
    }
}

File: app/lib/RecipeBasic.dart
class RecipeBasic {
  int id;
  String title;
  String image;
  int usedIngredientCount;
  int missedIngredientCount;
  int likes;

  RecipeBasic({this.id, this.title, this.image, this.usedIngredientCount, this.missedIngredientCount, this.likes});

  factory RecipeBasic.fromJson(Map<String, dynamic> json) {
    return RecipeBasic(
      id: json['id'],
      title: json['title'],
      image: json['image'],
      usedIngredientCount: json['usedIngredientCount'],
      missedIngredientCount: json['missedIngredientCount'],
      likes: json['likes']
    );
  }
}
